src/simulation.hpp

```diff
@@ -576,6 +577,14 @@ template <typename problem_t> void AMRSimulation<problem_t>::setInitialCondition
 		ReadCheckpointFile();
 	}
 
+	phi.resize(finest_level + 1);
+	const int nghost = 1;
+	const int ncomp = 1;
+	for (int lev = 0; lev <= finest_level; ++lev) {
+		phi[lev].define(grids[lev], dmap[lev], ncomp, nghost);
+		phi[lev].setVal(0); // set initial guess to zero
```

Comment:
`phi` will be zero in the `plt00000` output. Is that what you want?

Reply:
Umm I dont care about what it is in `plt00000`, but maybe other applications will? But how would we calculate it at t=0?

Reply:
You could split `ellipticSolveAllLevels` into two functions -- one that calculates `phi` by calling the `OpenBC` object, and another that adds the gravitational acceleration to the gas. Then you could call the first function here.

Reply:
So we need all of this part at t=0:

```
                for (int lev = 0; lev <= finest_level; ++lev) {
                        phi[lev].define(grids[lev], dmap[lev], ncomp, nghost);
                        rhs[lev].define(grids[lev], dmap[lev], ncomp, nghost);
                        phi[lev].setVal(0); // set initial guess to zero
                        rhs[lev].setVal(0);
                        fillPoissonRhsAtLevel(rhs[lev], lev);
                        rhs_min = std::min(rhs_min, rhs[lev].min(0));
                }

                amrex::Real abstol = abstolPoisson_ * rhs_min;
                poissonSolver.solve(amrex::GetVecOfPtrs(phi), amrex::GetVecOfConstPtrs(rhs), reltolPoisson_, abstol);
                if (verbose) {
                        amrex::Print() << "\n";
                }
```
?

Reply:
Yes, that would fix it so the phi output is valid in the first plotfile.